autopush/endpoint.py

```diff
@@ -584,11 +584,21 @@ def _route_notification(self, version, result, data, ttl=None):
     def _router_completed(self, response, uaid_data, warning=""):
         """Called after router has completed successfully"""
         # TODO: Add some custom wake logic here
-
         # Were we told to update the router data?
-        if response.router_data:
+        if response.router_data is not None:
+            if not response.router_data:
+                # An empty router_data object indicates that the record should
+                # be deleted. There is no longer valid route information for
+                # this record.
+                self.ap_settings.router.drop_user(self.uaid)
```

Comment:
I was wondering about that. In theory, we're already in a deferred thread from _route_notification.
I'll add a defer anyway.